batch/batch/worker/worker.py

```diff
@@ -137,6 +138,116 @@ def free(self, port):
         self.ports.put_nowait(port)
 
 
+class NetworkNamespace:
+    def __init__(self, subnet_index: int, private: bool):
+        self.private = private
+        self.network_name = uuid.uuid4().hex[:5]
+        self.hostname = uuid.uuid4().hex[:10]
+
+        if private:
+            self.host_ip = f'10.0.{subnet_index}.10'
+            self.job_ip = f'10.0.{subnet_index}.11'
+        else:
+            self.host_ip = f'10.1.{subnet_index}.10'
+            self.job_ip = f'10.1.{subnet_index}.11'
+
+        self.port = None
+        self.host_port = None
+
+    async def init(self):
+        veth_host = self.network_name + '-host'
+        veth_job = self.network_name + '-job'
+        await check_shell(
+            f'''
+ip netns add {self.network_name} && \
+ip link add name {veth_host} type veth peer name {veth_job} && \
+ip link set dev {veth_host} up && \
+ip link set {veth_job} netns {self.network_name} && \
+ip address add {self.host_ip}/24 dev {veth_host}
+ip -n {self.network_name} link set dev {veth_job} up && \
+ip -n {self.network_name} link set dev lo up && \
+ip -n {self.network_name} address add {self.job_ip}/24 dev {veth_job} && \
+ip -n {self.network_name} route add default via {self.host_ip}'''
+        )
+
+        await check_shell(
+            f'''
+iptables -w 10 --append FORWARD --in-interface {veth_host} --jump ACCEPT && \
+iptables -w 10 --append FORWARD --out-interface {veth_host} --jump ACCEPT'''
```

Comment:
This is for communicating with the internet. We create `veth_host`, which is the end of the job's veth pair in the *host* network namespace, but there is no connection yet between the internet-enabled "physical" device on the host and this device, so this forwards traffic between the two. It would be better if we specified the physical device in these forwarding rules, and I think that would do a better job of ensuring job devices can't speak to each other, but I haven't looked into programmatically getting the right device name.

Reply:
I just checked all the currently live batch instances and everyone has an ens5 and an lo. ens5 == PCI-Express slot 5. I'd bet Google's hypervisor always connects the ethernet device on slot 5. If this changes, we lose network connectivity, but its not a security risk.

Reply:
Should we just hard code `ens5` then? We can do a sanity check at worker startup time that `ens5` exists with `psutil`.